bin/pasta_show_maintainers.py

```diff
@@ -0,0 +1,165 @@
+
+    args = parser.parse_args()
+
+    if args.file!= None:
+        all_maintainers_text = file_to_string(args.file)
+    else:
+        all_maintainers_text = file_to_string('./resources/linux/repo/MAINTAINERS')
+    
+    if args.filter!= None:
+        filenames = file_to_string(args.filter).splitlines()
+
+    all_maintainers = LinuxMaintainers(all_maintainers_text)
+
+    if args.query == 'entries':
+        title = 'Entry'
+        counts, dirs= get_entry_counts(all_maintainers)
+        if args.filter != None:
+            counts_filt, dirs_filt = get_entry_counts(all_maintainers, filenames)
+    elif args.query == 'maintainers':
+        title = 'Maintainer'
+        counts, maintainers_subs = get_maintainer_counts(all_maintainers)
+        if args.filter != None:
+            counts_filt, _ = get_maintainer_counts(all_maintainers, filenames)
+    else:
+        return -1
+
+    if args.smallstat:
+        if args.filter != None:
+            results = get_small(title, counts_filt)
+        else:
+            results = get_small(title, counts)
+    else:
+        results = list()
+        if args.filter != None and args.query == 'entries':
+            results.append([title, 'Status', 'LoC in list', 'Total LoC', 'Byte count',  'In list(%)', 'Other directories'])
+            for item in counts_filt:
+                if any(all_maintainers.subsystems[item].status):
+                    results.append([item, all_maintainers.subsystems[item].status[0].value,counts_filt[item][0], counts[item][0], counts_filt[item][1], 
+                    round((counts_filt[item][0]/counts[item][0])*100, 2), ', '.join(dirs[item].difference(dirs_filt[item]))]) 
+                else:
+                    results.append([item, '', counts_filt[item][0], counts[item][0], counts_filt[item][1], round((counts_filt[item][0]/counts[item][0])*100, 2)])
+        elif args.filter != None and args.query == 'maintainers':
+            results.append([title, 'LoC in list', 'Total LoC', 'Byte count',  'In list(%)', 'Entries'])
+            for item in counts_filt:
+                results.append([item, counts_filt[item][0], counts[item][0], counts_filt[item][1], round((counts_filt[item][0]/counts[item][0])*100, 2), ', '.join(maintainers_subs[item]) ])
+        elif args.query == 'entries':
+            results.append([title, 'Status', 'LoC', 'Byte count'])
+            for item in counts:
+                if any(all_maintainers.subsystems[item].status):
+                    results.append([item, all_maintainers.subsystems[item].status[0].value, counts[item][0], counts[item][1]]) 
+                else:
+                    results.append([item, '', counts[item][0], counts[item][1]])
+        else:
+            results.append([title, 'LoC', 'Byte count', 'Entries'])
+            for item in counts:
+                results.append([item, counts[item][0], counts[item][1], ', '.join(maintainers_subs[item])])
+        
```

Comment:
you can use a dictionary with lambda functions, and then rewrite this code, so that the four cases are actually just the assignment of four different lists.

Reply:
and a generic part that handles the rest.